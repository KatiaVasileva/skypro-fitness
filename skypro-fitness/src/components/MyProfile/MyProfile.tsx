import { useNavigate } from "react-router-dom";
import { AppRoutes } from "../../lib/appRoutes";
import { useUserContext } from "../../hooks/useUserContext";

export const MyProfile = ({setIsOpen} : {setIsOpen: (isOpen: boolean) => void}) => {
  const { user, logout } = useUserContext();
  const navigate = useNavigate();

  const handleMyProfileButton = () => {
    navigate(AppRoutes.ACCOUNT_PAGE);
  };

  const handleLogoutButton = () => {
    logout();
    navigate(AppRoutes.MAIN);
    setIsOpen(false);
  };

  return (
    <div className="">
      <div className="block overflow-x-hidden absolute z-50 top-[70px] left-[-100px]">
        <div className="bg-white rounded-[30px] w-[266px] min-h-[258px] p-[30px] flex flex-col items-center shadow-md">
          <div className="mb-8 text-lg font-normal">
            <h3 className="">{user ? user.email : ""}</h3>
            <p className="text-gray">{user ? user.email : ""}</p>
          </div>

          <button
            className="btn-primary w-52 h-[52px] mb-[10px]"
            onClick={handleMyProfileButton}
          >
            Мой профиль
          </button>
          <button
            className="btn-primary-white w-52 h-[52px]"
            onClick={handleLogoutButton}
          >
            Выйти
          </button>
        </div>
      </div>
    </div>
  );
};
