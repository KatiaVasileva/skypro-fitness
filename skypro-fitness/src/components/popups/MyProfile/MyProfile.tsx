import { useNavigate } from "react-router-dom";
import { useUserContext } from "../../../hooks/useUserContext";
import { AppRoutes } from "../../../lib/appRoutes";

export const MyProfile = ({setIsOpen} : {setIsOpen: (isOpen: boolean) => void}) => {
  const { user, logout } = useUserContext();
  const navigate = useNavigate();

  const handleMyProfileButton = () => {
    navigate(AppRoutes.ACCOUNT);
  };

  const handleLogoutButton = () => {
    logout();
    navigate(AppRoutes.MAIN);
    setIsOpen(false);
  };

  return (
    <div className="">
      <div className="block overflow-x-hidden absolute z-50 top-[50px] md:top-[70px] left-[-190px] md:left-[-100px]">
        <div className="bg-white rounded-[30px] w-[266px] min-h-[258px] p-[30px] flex flex-col items-center shadow-md">
          <div className="mb-8 text-lg font-normal">
            <h3 className="text-center">{user ? user.displayName : ""}</h3>
            <p className="text-gray">{user ? user.email : ""}</p>
          </div>

          <button
            className="btn-primary w-52 h-[52px] mb-[10px]"
            onClick={handleMyProfileButton}
          >
            Мой профиль
          </button>
          <button
            className="btn-primary-white text-black w-52 h-[52px]"
            onClick={handleLogoutButton}
          >
            Выйти
          </button>
        </div>
      </div>
    </div>
  );
};
