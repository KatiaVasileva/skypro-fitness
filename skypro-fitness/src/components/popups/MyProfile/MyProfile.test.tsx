/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import * as router from "react-router";
import { BrowserRouter } from "react-router-dom";
import UserProvider from "../../../context/UserContext";
import CoursesProvider from "../../../context/CoursesContext";
import WorkoutProvider from "../../../context/WorkoutContext";
import { MyProfile } from "./MyProfile";

const navigate = jest.fn();

describe("MyProfile component", () => {
  beforeEach(() => {
    jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
  });

  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <CoursesProvider>
            <WorkoutProvider>
              <MyProfile setIsOpen={function (): void {
                            throw new Error("Function not implemented.");
                        } } />
            </WorkoutProvider>
          </CoursesProvider>
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
