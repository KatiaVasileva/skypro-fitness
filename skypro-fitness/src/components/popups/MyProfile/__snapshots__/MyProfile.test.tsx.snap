// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component renders correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="block overflow-x-hidden absolute z-50 top-[50px] md:top-[70px] left-[-190px] md:left-[-100px]"
    >
      <div
        class="bg-white rounded-[30px] w-[266px] min-h-[258px] p-[30px] flex flex-col items-center shadow-md"
      >
        <div
          class="mb-8 text-lg font-normal"
        >
          <h3
            class="text-center"
          />
          <p
            class="text-gray"
          />
        </div>
        <button
          class="btn-primary w-52 h-[52px] mb-[10px]"
        >
          Мой профиль
        </button>
        <button
          class="btn-primary-white text-black w-52 h-[52px]"
        >
          Выйти
        </button>
      </div>
    </div>
  </div>
</div>
`;
